stackslib/src/chainstate/stacks/boot/pox-4.clar

```diff
@@ -658,12 +659,17 @@
           ;; return the lock-up information, so the node can actually carry out the lock.
           (ok { stacker: tx-sender, lock-amount: amount-ustx, unlock-burn-height: (reward-cycle-to-burn-height (+ first-reward-cycle lock-period)) }))))
 
+;; Revokes the delegation to the current stacking pool.
+;; New in pox-4: Fails if the delegation was already revoked.
+;; Returns the last delegation state even if the delegation is already expired.
 (define-public (revoke-delegate-stx)
-  (begin
+  (let ((stacker { stacker: tx-sender })
+        (last-delegation-state (map-get? delegation-state stacker)))
```

Comment:
Should we use `get-check-delegation` instead of map-get and fail also if the delegation already expired?

Reply:
That makes sense to me, one less odd condition that event-observers have to deal with

Reply:
However, that implies that the map entry for expired delegation might never be deleted.

Reply:
Did they get deleted prior to this PR if users never performed a `revoke-delegate-stx`?

Reply:
Applied the change so that revoke-delegate-stx also fails after expiry

Reply:
Today, they are not deleted if the user does not perform a revoke. However, with this change, the user does not have a chance to remove the map entry once it is expired. 

I think it does not matter.